### ISSUE TITLE
Publishing pictures directly to Blogger

### ISSUE THREAD

**Issue by Dinchy87** [OpenLiveWriter/OpenLiveWriter#238]
I would like to say thanks for this work so far on Open Live Writer you guys are great and wanted to ask if we could somehow implement the posting of pictures to the same servers the blogger post does. I mean publishing directly from the website blogger.com gives your pictures an url like http://1.bp.blogspot.com/-59y4fvfzbsQ/VniBEzm_EII/AAAAAAAAV2w/Mtz4eP7eaVA/s1600/************.jpeg

where the same picture uploaded via OLW give an url to the user content of this article writer and post it directly to Google pictures. The url is then https://lh3.googleusercontent.com/-akb6pnucPtM/Vns3L3ePOGI/AAAAAAAAV-k/7CQWontQplQ/s1600-h/******* 

The issue i currently have with this is that pictures that are posted to Blogger are rendered properly in a WebView inside a WP8.1 WinRT app where the pictures that are posted from OLw and prior also WLW are posted to Google picasa/pictures and are not rendered at the first try (somehow).

Both of the above pictures are at the same resolution and both the same quality and everything but i get this nasty bug inside a webview where a third or soemthimes even 2/3 of the picture is not rendered at all. with pictures posted directly from Blogger i dont have this issue.

So is it possible that we can get somehow to post our pictures directly to blogger/blogspot servers?!

**Comment by ScottIsAFool** [comment]
I think this is possibly a dupe of #227

**Comment by martinwoodward** [comment]
Definitely related to #227 but not a total dupe. Should keep open for now.  I'm having trouble finding anything in the [Blogger API](https://developers.google.com/blogger/docs/3.0/reference/) related to uploading images which is why we kept using the old [Picasa web service](https://developers.google.com/picasa-web/code) that Live Writer has always used.

If someone can see the API that we should be using to post images directly to Blogger then it shouldn't be too much work to switch to that.

**Comment by Lozula** [comment]
I did some research into this a while ago because I was getting frustrated with WLW not letting me post images bigger than 800px wide in my blog posts, something I could do if I used the blogger post editor.

Basically, all blogger images are hosted on Picasa web albums, even when uploaded through the blogger interface. 

The difference is that third party apps have to use the google picasa web albums api to upload their photos, which places restrictions on images - you can't embed these images into web pages at resolutions higher than 800px wide for example. Blogger itself doesn't have these restrictions when uploading images directly through their post editor, but there doesn't appear to be an API available for uploading images to blogger other than the Picasa albums one.

**Comment by Dinchy87** [comment]
@Lozula yes that was my tought too that is restricted just to the web Interface of Blogger and third Party app dont have Access to this. I see this in my own app too the Pictures loaded in the post from Blogger directly open much faster with a higher Quality then the ones published via WLW or OLW. This is not a bug, ist always been there, just something i would like to see if ist possible. My WebView inside my WP8.1 really struggles with the Pictures published via app. While the ones posted from Blogger come crisp and clear to the webview. Its 100% related to this because i have tested it out a few days posting just to Blogger and a a Long time posted only on WLW/OLW.

It wold not be a Problem is the Pictures posted from OLW would come 100% trough, sometimes it just renders them to the half and the upper part is pixelated so much that you cannot see what it is. This does never happen with Blogger Pictures. This is why i opened this thread if ist possible to do Uploads directly like Blogger it would be a huge benefit like your post too proves that this pictures take too long to load. Maybe it is the same "issue" we have here.

**Comment by Dinchy87** [comment]
Its me again , i have looked at my picasa album where the pictures are stored from OLW and something nice, it seems that it adds the pictures in full res + the resolution we added in OLW. for me this is aprox 850px in width. both are above the Blogger "limit". So blogger uses tis then and scales them down with the ending ?imgmax=800 but we can manipulate this i think :) if you add the width of the picture you set in there like 850 for my pictures it scales them up :) as Picasa has the original picture too. Can you use this to make the pictures the width we set it in OLW ?! i think the same would happen if you set in OLW 1200px and you get the link end like ?imgmax=800, setting it to 1200 would give you the width you want.

**Comment by nvella** [comment]
For anyone that finds this issue, the current state of affairs is that Google still has no API for uploading images directly to Blogger. It is unclear if they will ever produce one. As a workaround, image upload to Google Drive has been implemented (#810) instead.

**Comment by shanselman** [comment]
Thanks for reporting this and for helping Open Live Writer over the years. We are doing an issue-bankruptcy cleanup so we can make the repository maintainable again and focus on current, actionable work. This older issue is being closed as part of that cleanup, not because it was unimportant. If this still reproduces in the current code or is still relevant to the revitalization work, please open a fresh issue with current details and we will triage it going forward.
